batch/batch/worker/worker_api.py

```diff
@@ -10,13 +13,46 @@
 from .disk import CloudDisk
 
 CredsType = TypeVar("CredsType", bound=CloudUserCredentials)
+ContainerCredentials = TypeVar("ContainerCredentials")
 
 
 class ContainerRegistryCredentials(TypedDict):
     username: str
     password: str
 
 
+class HailMetadataServer(abc.ABC, Generic[CredsType, ContainerCredentials]):
```

Comment:
Ah, OK, this took me a second to figure out. So there's a CredsType which is the Worker-owned-on-behalf-of-user credentials and there's `ContainerCredentials` which is just some internal representation used by the metadata server shim. I feel like the latter should be called "MetadataServerShimCredentials" or something?

Reply:
I also find having these multiple credential classes confusing, I'm going to take a stab at eliminating them and just using the `hailtop` credential classes.